scripts/Muon/GUI/Common/phase_table_widget/phase_table_presenter.py

```diff
@@ -0,0 +1,151 @@
+        self.view = view
+        self.context = context
+
+        self.group_change_observer = GenericObserver(self.update_current_groups_list)
+        self.run_change_observer = GenericObserver(self.update_current_run_list)
+        self.instrument_changed_observer = GenericObserver(self.update_current_phase_tables)
+
+        self.phase_table_calculation_complete_notifier = Observable()
+
+        self.update_current_phase_tables()
+
+    def update_view_from_model(self):
+        for key, item in self.context.phase_context.options_dict.items():
+            setattr(self.view, key, item)
+
+    def update_model_from_view(self):
+        for key in self.context.phase_context.options_dict:
+            self.context.phase_context.options_dict[key] = getattr(self.view, key, None)
+
+    def handle_calulate_phase_table_clicked(self):
+        self.update_model_from_view()
+
+        self.calculation_thread = self.create_calculation_thread()
+
+        self.calculation_thread.threadWrapperSetUp(self.handle_calculation_started, self.handle_calculation_success,
+                                                   self.handle_calculation_error)
+
+        self.calculation_thread.start()
+
+    def create_calculation_thread(self):
+        self._calculation_model = ThreadModelWrapper(self.calculate_phase_table)
+        return thread_model.ThreadModel(self._calculation_model)
+
+    def handle_calculate_phase_quad_button_clicked(self):
+        self.update_model_from_view()
+
+        self.phasequad_calculation_thread = self.create_phase_quad_calculation_thread()
+
+        self.phasequad_calculation_thread.threadWrapperSetUp(self.handle_calculation_started, self.handle_calculation_success,
+                                                             self.handle_calculation_error)
+
+        self.phasequad_calculation_thread.start()
+
+    def create_phase_quad_calculation_thread(self):
+        self._phasequad_calculation_model = ThreadModelWrapper(self.calculate_phase_quad)
+        return thread_model.ThreadModel(self._phasequad_calculation_model)
+
+    def calculate_phase_quad(self):
+        parameters = self.get_parameters_for_phase_quad()
+
+        phase_quad = run_PhaseQuad(parameters)
+
+        self.add_phase_quad_to_ADS(parameters, phase_quad)
+
+    def get_parameters_for_phase_quad(self):
+        parameters = {}
+        if self.context.phase_context.options_dict['phase_table_for_phase_quad'] == 'Construct':
+            parameters['PhaseTable'] = self.calculate_phase_table()
+        else:
+            parameters['PhaseTable'] = self.context.phase_context.options_dict['phase_table_for_phase_quad']
+
+        parameters['InputWorkspace'] = self.context.phase_context.options_dict['phase_quad_input_workspace']
+
+        return parameters
+
+    def add_phase_quad_to_ADS(self, parameters, phase_quad):
+        phase_quad_name = parameters['InputWorkspace'].split('_')[0] + '_PhaseQuad_phase_table_'\
```

Comment:
should this name generation go into the ADS handler?

Reply:
How names were generated was changed in #25573 so could we address this there?